File: TaskTracker.Model/TaskList/Request/GetTaskListRequest.cs
﻿using TaskTracker.Model.Attribute;

namespace TaskTracker.Model.TaskList.Request
{
    public class GetTaskListRequest : PagedRequest
    {
        public long? CreatorId { get; set; }
        [Join]
        public bool? JoinTasks { get; set; }
        [Join]
        public bool? JoinStatusGroup { get; set; }
        public long? SpaceId { get; set; }
    }
}

File: task-tracker-ui/src/app/model/task-list/GetTaskListRequest.ts
import { HttpParams } from "@angular/common/http";
import PagedRequest, { mapPagedRequest } from "../PagedRequest";

export default interface GetTaskListRequest extends PagedRequest {
    creatorId?: number,
    joinTasks?: boolean,
    joinStatusGroup?: boolean
}

export const mapGetTaskListRequest = (request: GetTaskListRequest): HttpParams => {
    let params = mapPagedRequest(request)

    if (request.creatorId) {
        params = params.append('CreatorId', request.creatorId)
    }
    if (request.joinTasks) {
        params = params.append('JoinTasks', request.joinTasks)
    }
    if (request.joinStatusGroup) {
        params = params.append('JoinStatusGroup', request.joinStatusGroup)
    }

    return params
}